### PR TITLE
Add Storybook case(Time Distance, Icon, Avatar,  Nameplate,Type Echo,  Click Boundary, Spinner Button, Code Block, Loading, Idea Popover)

### LINKED ISSUE #32: Add storybook Case
Recently, we added Storybook to the project, and we also need to write Case for each component.

- [x] [Time Distance](https://github.com/idea2app/Idea-React/blob/master/source/TimeDistance.tsx) - #33
- [x] [Icon](https://github.com/idea2app/Idea-React/blob/master/source/Icon.tsx) - #33
- [x] [Avatar](https://github.com/idea2app/Idea-React/blob/master/source/Avatar.tsx) - #33
- [x] [Nameplate](https://github.com/idea2app/Idea-React/blob/master/source/Nameplate.tsx) - #33
- [x] [Type Echo](https://github.com/idea2app/Idea-React/blob/master/source/TypeEcho.tsx) - #33
- [x] [Click Boundary](https://github.com/idea2app/Idea-React/blob/master/source/ClickBoundary.tsx) - #33
- [x] [Spinner Button](https://github.com/idea2app/Idea-React/blob/master/source/SpinnerButton.tsx) - #33
- [ ] [Select](https://github.com/idea2app/Idea-React/blob/master/source/Select.tsx)
- [x] [Code Block](https://github.com/idea2app/Idea-React/blob/master/source/CodeBlock.tsx) - #33
- [ ] [Editor](https://github.com/idea2app/Idea-React/blob/master/source/Editor.tsx)
- [ ] [Editor HTML](https://github.com/idea2app/Idea-React/blob/master/source/EditorHTML.tsx)
- [ ] [Open Map](https://github.com/idea2app/Idea-React/blob/master/source/OpenMap/index.tsx)
- [ ] [Table Spinner](https://github.com/idea2app/Idea-React/blob/master/source/TableSpinner.tsx)
- [x] [Loading](https://github.com/idea2app/Idea-React/blob/master/source/Loading.tsx) - #33
- [x] [Idea Popover](https://github.com/idea2app/Idea-React/blob/master/source/IdeaPopover.tsx) - #33
- [ ] [Idea Dialog](https://github.com/idea2app/Idea-React/blob/master/source/IdeaDialog.tsx)

### PR DESCRIPTION
**Pull request by stevending1st**
- [x]  [Time Distance](https://github.com/idea2app/Idea-React/blob/master/source/TimeDistance.tsx)
- [x]  [Icon](https://github.com/idea2app/Idea-React/blob/master/source/Icon.tsx)
- [x]  [Nameplate](https://github.com/idea2app/Idea-React/blob/master/source/Nameplate.tsx)
- [x]  [Type Echo](https://github.com/idea2app/Idea-React/blob/master/source/TypeEcho.tsx)
- [x]  [Click Boundary](https://github.com/idea2app/Idea-React/blob/master/source/ClickBoundary.tsx)
- [x]  [Spinner Button](https://github.com/idea2app/Idea-React/blob/master/source/SpinnerButton.tsx)
- [x]  [Code Block](https://github.com/idea2app/Idea-React/blob/master/source/CodeBlock.tsx)
- [x]  [Loading](https://github.com/idea2app/Idea-React/blob/master/source/Loading.tsx)
- [x]  [Idea Popover](https://github.com/idea2app/Idea-React/blob/master/source/IdeaPopover.tsx)

### CODE REVIEW

File: stories/CodeBlock.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export default meta;
+
+type Story = StoryObj<typeof CodeBlock>;
+
+export const Primary: Story = {
+    render: args => {
+        const { language, children } = args;
+        return <CodeBlock language={language}>{children}</CodeBlock>;
+    },
```
Comment (TechQuery) [inline stories/CodeBlock.stories.tsx]:
```suggestion
    render: ({ language, children }) => <CodeBlock language={language}>{children}</CodeBlock>,
```

File: stories/Icon.stories.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        size: 2,
+        className: 'shadow'
+    }
+};
+
+export const WithStyle: Story = {
+    args: {
+        ...Primary.args,
+        style: { border: '4px double #32a1ce' },
+        onClick: () => console.log('Hello World')
```
Comment (TechQuery) [inline stories/Icon.stories.ts]:
```suggestion
```
不推荐在图标上直接加点击事件，在外面包一层按钮比较语义化。

File: stories/IdeaDialog.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { IdeaDialog } from '../source/IdeaDialog';
+import { useState } from 'react';
```
Comment (TechQuery) [inline stories/IdeaDialog.stories.tsx]:
```suggestion
import type { Meta, StoryObj } from '@storybook/react';
import { useState } from 'react';

import { IdeaDialog } from '../source/IdeaDialog';
```

File: stories/IdeaDialog.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        const { children, onCancel, ...otherArgs } = args,
+            [showDialog, setShowDialog] = useState<boolean>(false);
+        return (
+            <>
+                <button onClick={() => setShowDialog(true)}> view </button>
+                <IdeaDialog
+                    show={showDialog}
+                    centered
+                    onCancel={() => setShowDialog(false)}
+                    {...otherArgs}
```
Comment (TechQuery) [inline stories/IdeaDialog.stories.tsx]:
```suggestion
    render: ({ children, onCancel, ...otherArgs }) => {
        const [showDialog, setShowDialog] = useState(false);

        return (
            <>
                <button onClick={() => setShowDialog(true)}>view</button>

                <IdeaDialog
                    show={showDialog}
                    centered
                    onCancel={() => setShowDialog(false)}
                    {...otherArgs}
```
有默认值应该是可以推导类型的

File: stories/IdeaPopover.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+type Story = StoryObj<typeof IdeaPopover>;
+
+export const Primary: Story = {
+    render: args => {
+        const { title, onShow } = args;
+        return (
+            <IdeaPopover onShow={onShow} title={title}>
+                <button>view</button>
+                <>This is the content of popover.</>
```
Comment (TechQuery) [inline stories/IdeaPopover.stories.tsx]:
```suggestion
    render: props => (
            <IdeaPopover {...props}>
                <button>view</button>
                This is the content of popover.
```

File: stories/Select.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+export const Primary: Story = {
+    render: ({ value, onChange, ...otherArgs }) => {
+        const [selectValue, setValue] = useState<string>(value);
+        return (
+            <Select
+                key="999999"
+                value={selectValue}
+                onChange={item => {
+                    onChange?.(item);
+                    setValue(item);
```
Comment (TechQuery) [inline stories/Select.stories.tsx]:
```suggestion
                    setValue(item);
                    onChange?.(item);
```

File: stories/Select.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        menuVariant: { control: 'text' }
+    }
+};
+
+export default meta;
+
+type Story = StoryObj<typeof Select>;
+
+export const Primary: Story = {
+    render: ({ value, onChange, ...otherArgs }) => {
```
Comment (TechQuery) [inline stories/Select.stories.tsx:25]: `Story` 是不是一个可以传 props 的范型？

File: stories/SpinnerButton.stories.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+import { SpinnerButton } from '../source/SpinnerButton';
+
+const meta: Meta<typeof SpinnerButton> = {
+    title: 'Idea-React/SpinnerButton',
+    component: SpinnerButton,
+    tags: ['autodocs'],
+    argTypes: {
+        disabled: { control: 'boolean' },
+        animation: { control: 'select', options: ['border', 'grow'] }
```
Comment (TechQuery) [inline stories/SpinnerButton.stories.ts]:
```suggestion
        animation: { control: 'select', options: ['border', 'grow'] },
        disabled: { control: 'boolean' }
```

File: stories/SpinnerButton.stories.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        ...Primary.args,
+        loading: true
+    }
+};
+
+export const GrowLoading: Story = {
+    args: {
+        ...Primary.args,
+        loading: true,
+        animation: 'grow'
```
Comment (TechQuery) [inline stories/SpinnerButton.stories.ts]:
```suggestion
        animation: 'grow',
        loading: true
```

**Review by TechQuery — CHANGES_REQUESTED** [review]

File: stories/Select.stories.tsx (hunk 7 above)
Comment (stevending1st) [reply stories/Select.stories.tsx:25]: image

File: stories/Select.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { Select, Option } from '../source/Select';
+import { useState } from 'react';
```
Comment (TechQuery) [inline stories/Select.stories.tsx:4]: 你配一下排序插件吧

**Review by TechQuery — CHANGES_REQUESTED** [review]

File: stories/Select.stories.tsx (hunk 10 above)
Comment (stevending1st) [reply stories/Select.stories.tsx:4]: > 你配一下排序插件吧

计划下个 PR 配置 eslint

**Review by TechQuery — APPROVED** [review]

**Pull request merged**